### PR TITLE
Add Linux setup instructions and script

### PR DESCRIPTION
**Pull request by mfeniseycopes**
Added a full set of instructions for installing Linux and setting up the dev environment. I've restricted Linux installation to only Ubuntu to keep it simple and uniform. Dual-boot instructions are also included. Also included is a script to install most of the tools we use here at App Academy.

### CODE REVIEW

**Comment by wolverdude** [comment]
Oops. Didn't see this and created a duplicate PR (https://github.com/appacademy/dotfiles/pull/24). Sorry.

File: linux/dev_setup_script.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
+printf "\n\n====> Linux Development Setup <===="
```
Comment (wolverdude) [inline linux/dev_setup_script.sh]: Add a crunchbang here and make this file executable. Also take of the `.sh`.

File: linux/dev_setup_script.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+printf "\nEach group of packages wl be install one at a time. Follow the prompts to continue.\n"
```
Comment (wolverdude) [inline linux/dev_setup_script.sh]: Please use the same log function that is used in the other scripts in this repo.

File: linux/dev_setup_script.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+sudo apt-get install git
+git config --global color.ui true
+git config --global user.name "your name here"
+git config --global user.email your_email_here
+
+printf "\n\n====> Git install complete <====\n"
+read -p "Press enter to continue"
+clear
+
+## Node
```
Comment (wolverdude) [inline linux/dev_setup_script.sh]: We should use NVM if at all possible.

File: linux/dev_setup_script.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
+sudo apt-get install nodejs
+sudo apt-get install npm
+sudo npm i -g webpack # installs webpack globally
```
Comment (wolverdude) [inline linux/dev_setup_script.sh]: Don't install webpack explicitly. Instead install all the packages in the node dependencies file.

File: linux/dev_setup_script.sh [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+### install bundler to manage dependencies
+gem install bundler
+rbenv rehash
+
+printf "\n\n====> Ruby + Bundler install complete <====\n"
+read -p "Press enter to continue"
+clear
+
+## Rails
+gem install rails
```
Comment (wolverdude) [inline linux/dev_setup_script.sh]: Do not install any gems explicitly (except bundler). Just install all the gems in the Gemfile.

File: linux/dev_setup_script.sh [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+rbenv rehash
+
+printf "\n\n====> Rails install complete <====\n"
+read -p "Done installing. Press enter to continue"
```
Comment (wolverdude) [inline linux/dev_setup_script.sh]: The atom packages don't appear to be installed by this script.

File: linux/dev_setup_script.sh (hunk 1 above)
Comment (wolverdude) [inline linux/dev_setup_script.sh]: Actually, it should be called `install_requirements_linux` or something like that.

Comment (wolverdude) [inline linux/dev_setup_script.sh]: and it belongs in the `bin/` directory.

File: bin/install_requirements_linux [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+sudo apt-get install postgresql postgresql-contrib
+# creates admin role in psql for current user
+sudo -u postgres createuser "$USER" -s
+
+log "\n\n====> Postgresql install complete <====\n"
+read -p "Press enter to continue"
+clear
+
+## Rbenv
```
Comment (wolverdude) [inline bin/install_requirements_linux]: Everything from here down should be shared with the mac installer, except maybe installing NVM

File: bin/install_requirements_linux [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+export PATH="$HOME/.rbenv/bin:$PATH"
+export eval "$(rbenv init -)"
+source ~/.bashrc
+git clone https://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build
+
+log "\n\n====> Rbenv install complete <====\n"
+read -p "Press enter to continue"
+clear
+
+## Ruby
```
Comment (wolverdude) [inline bin/install_requirements_linux]: Everything from here down should be shared with the mac installer, except maybe installing NVM

File: bin/install_requirements_linux [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+fail() {
+  log -e "${RED}${script} failed:${RESET} $1" >&2
+  exit 1
+}
+
+cd "$( dirname "$0" )/../requirements"
+requirements_dir="$( pwd )"
+
+log "\n\n====> Linux Development Setup <===="
```
Comment (wolverdude) [inline bin/install_requirements_linux]: please keep the logging format consistent

File: bin/install_requirements_linux [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+log "\nEach group of packages will be install one at a time. Follow the prompts to continue.\n"
+read -p "Press enter to continue"
```
Comment (wolverdude) [inline bin/install_requirements_linux]: I think this stopping periodically is good practice, just to make sure students understand what's going on. Let's port these over to the other install files.

File: bin/install_requirements [hunk 11]
```diff
@@ -1,11 +1,10 @@
 #!/bin/bash
-RUBY_VERSION='2.3.1'
-NODE_VERSION='6.2.1'
 
 RED='\033[0;31m'
 PURPLE='\033[0;35m'
 RESET='\033[0m'
 script="$(basename $0)"
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (wolverdude) [inline bin/install_requirements]: User `$0`

File: bin/install_requirements [hunk 12]
```diff
@@ -1,11 +1,10 @@
 #!/bin/bash
-RUBY_VERSION='2.3.1'
-NODE_VERSION='6.2.1'
 
 RED='\033[0;31m'
 PURPLE='\033[0;35m'
 RESET='\033[0m'
 script="$(basename $0)"
```
Comment (wolverdude) [inline bin/install_requirements]: put double quotes around `$0`

File: bin/install_requirements_linux [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+clear
+
+## Git
+sudo apt-get install git
+git config --global color.ui true
+git config --global user.name "your name here"
+git config --global user.email your_email_here
+
+# Git Completion
+git_url="https://raw.githubusercontent.com/git/git/master/contrib/completion"
```
Comment (wolverdude) [inline bin/install_requirements_linux]: this should be in the common script

File: requirements/node-packages-linux.txt [hunk 14]
```diff
@@ -0,0 +1,4 @@
+eslint
```
Comment (wolverdude) [inline requirements/node-packages-linux.txt]: Don't create this file. Just add `ws` to the main node-packages file

**Comment by mfeniseycopes** [comment]
Hey @niko79542, @wolverdude, thanks for putting so much work to get the Linux script merged. I think it will really help many of the students who don't have or can't afford Macs. 👍

**Pull request merged**